### ISSUE TITLE
Feature Request: Add regex profiling feature

### ISSUE THREAD

**Issue by dalemyers** [realm/SwiftLint#2971]
We have 29 different regex rules and suspect that they are part of the reason we are seeing significant performance issues. If there were a regex profiling mechanism, we could tell which rules are taking the longest to run and then tailor those explicitly to be as fast as possible.

**Comment by marcelofabri** [comment]
There's an argument `--benchmark` which outputs times per rule. However, I think that custom rules are grouped together. While not ideal, you could change your configuration and run it with different sets of custom rules and compare the values.

**Comment by dalemyers** [comment]
That's good to know (I don't believe there's any way to see these arguments without looking at the source, right?). 

With custom rules grouped together, changing it isn't much different than just modifying the swiftlint.yml file and running `time swiftlint` and comparing the differences between runs. Something a little more involved would be helpful. 

For what it's worth, we did that and got the numbers we cared about, so I'm happy to let this one go, but I do see how it could be useful for others in the future.
